### DISCUSSION TITLE
Spacy model `en-core-web-trf` hangs when executed in a celery task

### DISCUSSION THREAD

**Discussion by linaran** [explosion/spaCy#9028] [Help: Other Questions] [answered]
### What is celery?
Celery is a standard python job queue meant to schedule and execute tasks at a later point.
A celery task is just a python function annotated `@task`.

### Bug description
Here's a short snippet of code I use to access `spacy` features from a celery task:
```python
import spacy
NLP = spacy.load("en_core_web_trf")

def get_people(text):
    sents = NLP(text)
    return [str(ee) for ee in sents.ents if ee.label_ == "PERSON"]
```

Here's a short snippet of code to reproduce the celery issue:
```python
@task
def testing_spacy():
    import spacy_utils # Snippet of code above

    print(spacy_utils.get_people("John called Mary"))
```

If the function from above, is called directly, i.e. `testing_spacy()`, everything will execute fine and print a result shortly.
If it's called as a celery task, i.e `testing_spacy.delay()`, it will never finish. I confirmed a worker is processing the task.

Furthermore, I came across a closed issue #6662 which described a similar bug. I ran that PoC (normally in a python shell), and it got stuck with this warning:
```
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
        - Avoid using `tokenizers` before the fork if possible
        - Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
```

Trying to set `TOKENIZERS_PARALLELISM` to `0/false` only removed the warning message but didn't fix the bug for the PoC.

I'm not sure what's the difference between running this code normally and in celery but the PoC from the mentioned issue also isn't working so I think spacy tries some parallelism that's available when running in celery.

## Your Environment

* Operating System: Ubuntu 18.04 (as a docker container on a Mac)
* Python Version Used: 3.6.9
* spaCy Version Used: 3.1.2
* spacy-transformers: 1.0.4
* en-core-web-trf: 3.1.0
Spacy was installed with `pip install spacy` and the model was downloaded with `python -m spacy download en_core_web_trf`.

Thanks for the help in advance, I hope I just misconfigured something basic.

**Comment by adrianeboyd** [comment, marked as answer]
Hi, please see: https://github.com/explosion/spaCy/discussions/8694

**Reply by linaran** [reply]
Sorry, didn't know github had discussions 😄 
I didn't try the `PoC` yet but your torch workaround seems to work.
I'll have to check what's the performance impact, but nevertheless thanks for providing a workaround.

EDIT: For some reason the `mark as answer` button isn't doing anything. I'll try to click it later again.

**Reply by adrianeboyd** [reply]
Yeah, the discussion board has a lot of nice features, but it definitely makes searching for related issues much harder. I wish github would offer a better combined search of issues+discussions.

**Reply by linaran** [reply]
Just a short follow-up, after implementing the workaround and benchmarking it, it turns out it's significantly better to simply downgrade to `en_core_web_lg`.

When recognizing people on texts up to 5000 chars the single-threaded variant is just too slow.
